app/stac_api/pagination.py

```diff
@@ -9,103 +9,185 @@
 from rest_framework.exceptions import ValidationError
 from rest_framework.request import Request
 from rest_framework.response import Response
+from rest_framework.utils.urls import remove_query_param
+from rest_framework.utils.urls import replace_query_param
 
 from stac_api.utils import get_query_params
 from stac_api.utils import remove_query_params
 
 logger = logging.getLogger(__name__)
 
 
+def update_links_with_pagination(data, previous_url, next_url):
+    '''Update the links dictionary with the previous and next link if needed
+
+    When no 'links' is present in data it is added even if there is no previous and/or next link
+    to add.
+
+    Args:
+        data: dict
+            data dictionary
+        previous_url: string
+            previous url
+        next_url: string
+            next url
+
+    Returns: tuple(dict, None|dict, None|dict)
+        Tuple (data, previous_link, next_link)
+    '''
+    links = []
+    previous_link = None
+    next_link = None
+    if next_url is not None:
+        next_link = {'rel': 'next', 'href': next_url}
+        links.append(next_link)
+    if previous_url is not None:
+        previous_link = {'rel': 'previous', 'href': previous_url}
+        links.append(previous_link)
+
+    if 'links' not in data:
+        data.update({'links': links})
+    elif links:
+        data['links'] += links
+    return data, previous_link, next_link
+
+
+def validate_page_size(size_string, max_page_size, log_extra=None):
+    '''Parse and validate page size
+
+    Args:
+        size_string: string
+            page size in string
+        max_page_size: int
+            max page size allowed
+        log_extra: dict
+            extra to add to the log message in case of error
+
+    Returns: int
+        Page size parsed
+
+    Raises:
+        ValidationError: if page size is invalid
+    '''
+    try:
+        page_size = int(size_string)
+    except ValueError as error:
+        logger.error(
+            'Invalid query parameter limit=%s: must be an integer', size_string, extra=log_extra
+        )
+        raise ValidationError(
+            _('invalid limit query parameter: must be an integer'),
+            code='limit'
+        ) from None
+
+    if page_size <= 0:
+        logger.error(
+            'Invalid query parameter limit=%d: negative number not allowed',
+            page_size,
+            extra=log_extra
+        )
+        raise ValidationError(
+            _('limit query parameter too small, must be in range 1..%d') % (max_page_size),
+            code='limit'
+        )
+    if max_page_size and page_size > max_page_size:
```

Comment:
wouldn't simply `if page_size > max_page-size` do here?

Reply:
`max_page_size` comes from `settings.REST_FRAMEWORK['PAGE_SIZE_LIMIT']` which could be None, that's why I'm checking it first.

Reply:
OK, I see, thanks. So it could be passes as None to this function. Probably might make sense to add `| None` in the docstring for `max_page_size`?